src/libsyntax/errors/emitter.rs

```diff
@@ -23,6 +22,23 @@ use std::io;
 use std::rc::Rc;
 use term;
 
+#[derive(Debug)]
+enum Error {
+    Io(io::Error),
+    Term(term::Error),
+}
+
+impl From<io::Error> for Error {
+    fn from(err: io::Error) -> Self {
+        Error::Io(err)
+    }
+}
+
+impl From<term::Error> for Error {
+    fn from(err: term::Error) -> Self {
+        Error::Term(err)
+    }
+}
```

Comment:
Yep, it does get a bit obnoxious to setup the first time, and if we never take advantage of it, it does get a bit silly. I'm fine either way, assuming if we add this impl to `term` it's fine with @Stebalien.